### PR TITLE
Bug 2069310: UPSTREAM: <carry>: use hardcoded rest mapper from library-go

### PR DESCRIPTION
**Pull request by tkashem**

### CODE REVIEW

**Comment by tkashem** [comment]
1.  If the Bugzilla associated with the PR has the "FastFix" keyword, the subjective assessment on the issue has already been done and a customer is impacted. These PRs should be prioritized for merge.
    - [ ] verified
    - [x] does not apply
2. The bug has significant impact either through severity, reduction in supportability, or number of users affected.
    - [x] verified
    - [ ] does not apply
3. For branches that are in the Maintenance lifecycle phase:
    - [ ] The bug is a critical fix, no reasonable workaround exists, and a recommendation for upgrade has been ruled out, or 
    - [ ] The bug is a security related bug
    - [x] Branch not in maintenance mode yet (current release + previous release for 90 days after current GA; everything older is in maintenance)
4. The severity field of the bug must be set to accurately reflect criticality.
    - [x] verified
5. The PR was created with the cherry-pick bot OR the PR’s description is well formed with user-focused release notes that state the bug number, impact, cause, and resolution. Where appropriate, it should also contain information about how a user can identify whether a particular cluster is affected.
    - [x] verified

**Comment by aojea** [comment]
@sosiouxme is 4.9 using golang 1.17 or 1.18 without the ENV variable hack ?

**Comment by sosiouxme** [comment]
> is it possible that the builder image doesn't have the correct go version? the patch seems to be there

```
🐚 podman run -it registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.9 go version
go version go1.16.12 linux/amd64
```

that's the updated version which I built that should have the patch using the build flag. I didn't know a good way to test that it was correctly allowing the deprecated format, so I was relying on the unit tests to tell us...

File: go.mod [hunk 1]
```diff
@@ -70,7 +70,7 @@ require (
 	github.com/openshift/api v0.0.0-20211028023115-7224b732cc14
 	github.com/openshift/apiserver-library-go v0.0.0-20211105091019-06e87e7030eb
 	github.com/openshift/client-go v0.0.0-20210831095141-e19a065e79f7
-	github.com/openshift/library-go v0.0.0-20211207082558-36821c7a1fa2
+	github.com/openshift/library-go v0.0.0-20220405121559-e304504b7d6f
```
Comment (p0lyn0mial) [inline go.mod:73]: okay, this points to https://github.com/openshift/library-go/commits/release-4.9

**Pull request merged**